### PR TITLE
Badge the Network tab when we receive failed network requests.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Screen Shot 2020-12-22 at 2 40 25 PM
Part of https://github.com/flutter/devtools/issues/173

### CODE REVIEW

File: packages/devtools_app/lib/src/network/network_controller.dart [hunk 1]
```diff
@@ -343,26 +347,45 @@ class NetworkController
             final caseInsensitiveSubstring = substring.toLowerCase();
             final matchesUri =
                 r.uri.toLowerCase().contains(caseInsensitiveSubstring);
-            if (matchesUri) return true;
+            if (matchesUri) {
```
Comment (jacob314) [inline packages/devtools_app/lib/src/network/network_controller.dart:350]: these four blocks all look the same.
Maybe a helper that takes `bool checkMatch(r, textToMatch, caseInsensitiveSubstring)`

Comment (jacob314) [reply packages/devtools_app/lib/src/network/network_controller.dart:350]: alternately just run
filteredData.value.forEach(checkForError)
after building up the filteredData. That seems a bit easier to get all the cases right.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/network/network_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/network/network_controller.dart]: done

**Pull request merged**